tests/tests_mocking_pyfirecrest/test_transport.py

```diff
@@ -0,0 +1,627 @@
+def test_getfile(firecrest_computer: orm.Computer, tmpdir: Path):
+    transport = firecrest_computer.get_transport()
+
+    _remote = tmpdir / "remotedir"
+    _local = tmpdir / "localdir"
+    _remote.mkdir()
+    _local.mkdir()
+
+    Path(_remote / "file1").write_text("file1")
+    Path(_remote / ".hidden").write_text(".hidden")
+    os.symlink(_remote / "file1", _remote / "file1_link")
+
+    # raise if remote file does not exist
+    with pytest.raises(FileNotFoundError):
+        transport.getfile(_remote / "does_not_exist", _local)
+
+    # raise if localfilename not provided
+    with pytest.raises(IsADirectoryError):
+        transport.getfile(_remote / "file1", _local)
+
+    # raise if localpath is relative
+    with pytest.raises(ValueError):
+        transport.getfile(_remote / "file1", Path(_local / "file1").relative_to(tmpdir))
+
+    # don't mix up directory with file
+    with pytest.raises(FileNotFoundError):
+        transport.getfile(_remote, _local / "file1")
+
+    # write where I tell you to
+    transport.getfile(_remote / "file1", _local / "file1")
+    transport.getfile(_remote / "file1", _local / "file1-prime")
+    assert Path(_local / "file1").read_text() == "file1"
+    assert Path(_local / "file1-prime").read_text() == "file1"
+
+    # always overwrite
+    transport.getfile(_remote / "file1", _local / "file1")
+    assert Path(_local / "file1").read_text() == "file1"
+
+    Path(_local / "file1").write_text("notfile1")
+
+    transport.getfile(_remote / "file1", _local / "file1")
+    assert Path(_local / "file1").read_text() == "file1"
+
+    # don't skip hidden files
+    transport.getfile(_remote / ".hidden", _local / ".hidden-prime")
+    assert Path(_local / ".hidden-prime").read_text() == ".hidden"
+
+    # follow links
+    transport.getfile(_remote / "file1_link", _local / "file1_link")
+    assert Path(_local / "file1_link").read_text() == "file1"
+    assert not Path(_local / "file1_link").is_symlink()
+
+
+@pytest.mark.parametrize("payoff", [True, False])
+@pytest.mark.usefixtures("aiida_profile_clean")
+def test_gettree(firecrest_computer: orm.Computer, tmpdir: Path, payoff: bool):
+    """
+    This test is to check `gettree` through non tar mode.
+    bytar= True in this test.
+    """
+    transport = firecrest_computer.get_transport()
+    transport.payoff_override = payoff
+
+    # Note:
+    # SSH transport behaviour, 69 is a directory
+    # transport.get('somepath/69', 'someremotepath/') == transport.get('somepath/69', 'someremotepath')
+    # transport.get('somepath/69', 'someremotepath/') == transport.get('somepath/69/', 'someremotepath/')
+    # transport.get('someremotepath/69', 'somepath/69')--> if 69 exist, create 69 inside it ('somepath/69/69')
+    # transport.get('someremotepath/69', 'somepath/69')--> if 69 no texist,create 69 inside it ('somepath/69')
+    # transport.get('somepath/69', 'someremotepath/6889/69') --> create everything, make_parent = True
```

Comment:
Can you maybe explain more why this is important here?  You don't use `get` here right?

Reply:
This is just to remember what sort of behavior `aiida-core` expects from a transport plugin.
This will still do, because if `source_path` is a directory `get()` with call `gettree()` internally.
But we can make it more clear, how about this:

```python
    # transport.gettree('somepath/69', 'someremotepath/') == transport.get('somepath/69/', 'someremotepath/')
    # transport.gettree('someremotepath/69', 'somepath/69')--> if 69 exist, create 69 inside it ('somepath/69/69')
    # transport.gettree('someremotepath/69', 'somepath/69')--> if 69 no texist,create 69 inside it ('somepath/69')
    # transport.gettree('somepath/69', 'someremotepath/6889/69') --> create everything, make_parent = True
```